### PR TITLE
fix: suggested changes by TA

### LINKED ISSUE #13: CL - Review
- Sidebar can be on the viewport, rn it covers the text when poped out
- Have a proper undo button in the alert.
- Have the code snippets with the same width in the documentation.
- Badge and float button needs to be a perfect circle
- Badge should be on the top and at the edge of the text/element. 
- Card: apply proper padding in cards, align the icons with text, remove margin from the badge on the card and give margin to close icon on the card.
- Apply proper padding in the toast/snack bar, increase the time toast stays on the screen.
- The toast should always be positioned on the viewport.
- The text utilities page not working
- class Naming at a few places is not explanatory, improve that and check in other files as well. Ex: .bold in alert.css
- The JS code for alert could have been reused.
- Consistency in units, if you are using px for less than 10px then use that everywhere.
- Extract display: flex and justify content and repetitive code to a separate class and use it everywhere. 
- Use rems/ems for cards, paddings, and margins
- Why use vh in margin?
- Use color variables everywhere in your code and make sure that the naming of color variables is consistent.
-  display: none should be used to hide alerts and not opacity.
- Apply styles to sliders. 

- UI Improvements: 
Documentation content can be in the center
Buttons for navigation should have a different background color to set them apart from normal background.
The Sidebar should always be present on desktop screens as navigation without it is a hassle.
Change the color of links to improve Readability. 
Use primary or some light color for the link button.
Avoid star(*) before username and password in input.
Why is a disabled button added in the radio and checkbox?
Change colors of CTA buttons in modal
Dismissable drawer sidebar should be present by default and hidden on a smaller screen size.
Use bigger icons in the Rating
Not to use hacky value ex: .0001, 1.382555
Remove console.log from the code.
Have primary color more visible to the eye than secondary color.

### PR DESCRIPTION
**Pull request by krgarima**
## Change Logs:
# Description: Fixed UI and code

# More details:

1. `Sidebar` -
- Sidebar opened permanently and closes only when app is opened on a small screen. It was earlier overlapping Documentation content but now I have shifted the whole content to the center.
- Buttons for navigation now have a different background color to set them apart from normal background in the sidebar.

2. `Alert` - 
- Reused the JS code for alert.

3. `Badges` - 
- Badges and float button made into a perfect circle.
- Adjusted badges. Few of them were not aligned properly.
- Replaced opacity with visibility property in the JS code.

4. `Buttons` - 
- Changed link button color to a light color.

5. `Cards` - 
- Aligned icons with text.
- Adjusted badge on the card.
- Gave a bit margin to 'close' icon on the card.

6. `Input` - 
- Removed star(*) before username and password in input.

7. `Modal` -
- Change colors of CTA buttons in modal.

8. `Navigation` - 
- Dismissable drawer sidebar opened permanently for large screens and hidden in case of a smaller screen size. 
- Changed hacky value ex: .0001, 1.382555.

9. `Rating` -
- Increased icon sizes.

10. `Toast/ Snackbar` -
- Applied proper padding in the toast/snack bar. Also, increased the time toast stays on the screen.
-  Positioned toast relative to the viewport

11. `Sliders` - 
- Added 1 more styled slider.

### Other suggestions - 
- Class Naming at a few places is not explanatory, improve that and check in other files as well. Ex: .bold in alert.css
- Consistency in units, if you are using px for less than 10px then use that everywhere.
- Extract display: flex and justify content and repetitive code to a separate class and use it everywhere. 
- Use rems/ems for cards, paddings, and margins. Not to use vh for margin.
- Use color variables everywhere in your code and make sure that the naming of color variables is consistent.

### CODE REVIEW

**Review by sakshi006 — APPROVED** [review]
1. Close me alert not getting closed.
2.  text-utilities component is not opening for me?

Looks good to merge

File: CSS/styleGuide.css [hunk 1]
```diff
@@ -30,6 +26,7 @@ html {
   --background-color: rgb(240 240 240);
   --text-color: #353434;
   --white-color: white;
+  --white2-color: floralwhite;
   --greenishWhite: rgb(225, 228, 226);
   --black-color: black;
```
Comment (ivinayakg) [inline CSS/styleGuide.css:31]: you don't need to define so many colors, as it's hard from a UX perspective to pull off so many colors, as most websites need at most 3-4 colors, and if you want to create shades of the colors there is a better approach, read More :-

https://stackoverflow.com/questions/40010597/how-do-i-apply-opacity-to-a-css-color-variable

File: Components/Alert/alert.css [hunk 2]
```diff
@@ -1,233 +1,241 @@
...
 
 /* simple alert */
 
-.btn__simple--primary,  .btn__outlined--primary{
-    padding: 1rem;
-    background-color: var(--blue-color);
-    color: var(--white-color);
-    border-radius: 5px;
+.btn__simple--primary,
+.btn__outlined--primary {
```
Comment (ivinayakg) [inline Components/Alert/alert.css:34]: create .btn__simple or .btn__outlined and then create separate classes like btn--primary that way you will reduce the no. of classes yu have to create

**Review by ivinayakg — APPROVED** [review]
next time, try batching up the changes if can, 
by the way, great work, well thought through, and structured code.

**Comment by krgarima** [comment]
> 1. Close me alert not getting closed.
> 
>     2. text-utilities component is not opening for me?
> 
> 
> Looks good to merge

1. Did not add the js code for close me.
2. Text utilties feature is still not added.

File: CSS/styleGuide.css (hunk 1 above)
Comment (krgarima) [reply CSS/styleGuide.css:31]: Thank you for the suggestion. I will read this.

File: Components/Alert/alert.css (hunk 2 above)
Comment (krgarima) [reply Components/Alert/alert.css:34]: right

**Pull request merged**